core/os/os.h

```diff
@@ -116,6 +118,8 @@ class OS {
 
 	virtual void initialize_logger();
 	virtual void initialize_core() = 0;
+	PhysicsServer *new_physics_server();
+	virtual void initialize_physics() = 0;
```

Comment:
Actually this is a really good idea, but I would just make this function non abstract and put the initialization code here, since it seems unlikely it may change on different OSs.

Reply:
That part still has to be done I think?